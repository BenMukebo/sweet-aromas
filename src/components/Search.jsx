import React from 'react';
import PropTypes from 'prop-types';

import { IconContext } from 'react-icons';
import { FiSearch } from 'react-icons/fi';

export default function Search(props) {
  const { query, onChange } = props;
  return (
    <div className="search-bar-main-container d-flex flex-center">
      <IconContext.Provider value={{ className: 'search-icon' }}>
        <FiSearch />
      </IconContext.Provider>
      <input type="text" value={query} onChange={onChange} className="search-input" placeholder="Search reciples..." />
    </div>
  );
}

Search.propTypes = {
  query: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
