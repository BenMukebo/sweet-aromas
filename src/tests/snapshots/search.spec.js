import renderer from 'react-test-renderer';

import Search from '../../components/Search';

describe('Search', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Search
          query="Beef"
          onChange={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
