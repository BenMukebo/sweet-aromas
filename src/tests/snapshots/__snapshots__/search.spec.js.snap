// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
<div
  className="search-bar-main-container d-flex flex-center"
>
  <svg
    className="search-icon"
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="11"
      cy="11"
      r="8"
    />
    <line
      x1="21"
      x2="16.65"
      y1="21"
      y2="16.65"
    />
  </svg>
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="Search recipes by name or area..."
    type="text"
    value="Beef"
  />
</div>
`;
